mon_school/mon_school/doctype/lms_sketch_like/lms_sketch_like.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2021, FOSS United and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+
+
+class LMSSketchLike(Document):
+    def after_insert(self):
+        sketch = self.get_sketch()
+        sketch.update_metrics()
+
+    def after_delete(self):
+        sketch = self.get_sketch()
+        sketch.update_metrics()
+
+    def get_sketch(self):
+        return frappe.get_doc("LMS Sketch", self.sketch)
```

Comment:
You may want to add `for_update=True` to avoid race conditions in updating metrics. That will lock the sketch and anyone else trying to run `for_update` query on the same row will be blocked until this transaction is complete.

Reply:
we'd have to lock the metrics doctype itself here, right? since that is what we'll be updating.

This may work anyways because getting the sketch would be the first step of that transaction. But I'm worried that it may block another `SELECT * FROM sketches` that a different user would run to get sketches. (I read this stackoverflow [answer](https://stackoverflow.com/a/16425603))

What do you think?

Reply:
Just tried it out with 2 connections to MariaDB. It will only block another `FOR UPDATE` query and not a normal select query.
This should work too :+1: